src/term/net_to_term.rs

```diff
@@ -109,7 +109,7 @@ pub fn net_to_term_non_linear(net: &INet, book: &Book) -> (Term, bool) {
             let (fst, fst_valid) = reader(net, fst, namegen, dup_scope, tup_scope, book);
             let (snd, snd_valid) = reader(net, snd, namegen, dup_scope, tup_scope, book);
             let valid = fst_valid && snd_valid;
-            (Term::Sup { fst: Box::new(fst), snd: Box::new(snd) }, valid)
+            (Term::Sup { tag: None, fst: Box::new(fst), snd: Box::new(snd) }, valid)
```

Comment:
Readback of dup tag here. We can check that this label is one of the ones decided by the user and for automatically decided labels we can use the number as the name (since numbers are not valid user-given names).